consumer/pdata/common.go

```diff
@@ -293,8 +297,28 @@ func (a AttributeValue) Equal(av AttributeValue) bool {
 		return v.IntValue == av.orig.GetIntValue()
 	case *otlpcommon.AnyValue_DoubleValue:
 		return v.DoubleValue == av.orig.GetDoubleValue()
+	case *otlpcommon.AnyValue_ArrayValue:
+		vv := v.ArrayValue.GetValues()
+		avv := av.orig.GetArrayValue().GetValues()
+		if len(vv) != len(avv) {
+			return false
+		}
+
+		for i, val := range avv {
+			val := val
+			av := newAttributeValue(&vv[i])
+
+			switch av.orig.Value.(type) {
+			case *otlpcommon.AnyValue_ArrayValue, *otlpcommon.AnyValue_KvlistValue:
+				return false
+			}
+
+			if !av.Equal(newAttributeValue(&val)) {
+				return false
+			}
+		}
+		return true
 	}
-	// TODO: handle MAP and ARRAY data types.
```

Comment:
You still need a TODO for the map correct?

Reply:
According to the spec, map isn't a valid AttributeValue, so I don't know if there should be equal for it

Reply:
I think we should have the TODO, not implement it :)

Reply:
Honestly, I thought someone had mistaken implementing map type. Okay, I'll return TODO

Reply:
Map is used for logs... Not for trace and resources 😟

@tigrannajaryan for more details

Reply:
Map value is not allowed by Otel API spec, but it is a valid value for our internal pdata and for OTLP. OTLP data model is a superset of Otel API data model. OTLP data model is wider so that it supports other reasonable telemetry formats not just Otel, so there is no contradiction here.
The TODO here is valid and we actually need to implement it eventually (it's just that we did not need it yet).